CMSgov/mint-app
# [EASI-2713] Email Notification:  Model Plan Dates Changed

# EASI-2713

## Changes and Description

- Add email notifications when model plan dates change



## How to test this change

- Run `scripts/dev down && scripts/dev up && scripts/dev db:seed`
- Assert that the following database table is created: `config.model_plan_date_changed_email_recipients`
- Assert that the database table contains the following test recipients: `one@test.gov, two@test.gov`
- Open `Mailcatcher` at `http://localhost:1085/`
- Navigate to the `Model basics` task list section in the model plan `Plan with Basics`
- Assign any `Model Category` selection
- Click `Next`
- Assert that no email is generated
  - Faulty Date Change processing would generate a data as `Plan Basics` is updated here
- Click `Next`
- Permutation Testing
  - Assign various different dates for testing purposes
    - There are two types of dates - individual dates and date ranges. Ensure they work both independently and together.
  - Assert that you receive an email in `Mailcatcher` with changes according to your test and that the `To` column contains all recipients in your `config.model_plan_date_changed_email_recipients` table (`one@test.gov, two@test.gov` by default)

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
